opm/simulators/flow/FlowMainEbos.hpp

```diff
@@ -343,33 +349,32 @@ namespace Opm
         }
     private:
         // called by execute() or executeInitStep()
-        int execute_(int argc, char** argv, bool output_cout, bool output_to_files,
-            int (FlowMainEbos::* runOrInitFunc)(bool), bool cleanup)
+        int execute_(int (FlowMainEbos::* runOrInitFunc)(), bool cleanup)
         {
             try {
                 // deal with some administrative boilerplate
 
-                int status = setupParameters_(argc, argv);
+                int status = setupParameters_(argc_, argv_);
                 if (status)
                     return status;
 
                 setupParallelism();
                 setupEbosSimulator();
-                runDiagnostics(output_cout);
+                runDiagnostics();
                 createSimulator();
 
                 // if run, do the actual work, else just initialize
-                int exitCode = (this->*runOrInitFunc)(output_cout);
+                int exitCode = (this->*runOrInitFunc)();
                 if (cleanup) {
-                    executeCleanup_(output_to_files);
+                    executeCleanup_();
                 }
                 return exitCode;
             }
             catch (const std::exception& e) {
                 std::ostringstream message;
                 message  << "Program threw an exception: " << e.what();
 
-                if (output_cout) {
+                if (output_cout_) {
```

Comment:
Cool if you use `this->output_cout_` to separate better between member variable and argument.

Reply:
Thanks for the quick review. Yes it is good to separate between local variables in the method and member variables of the class. I completely agree.  But... what about the underscore? I thought we used a trailing underscore to signify that a variable was a private member variable? So then it is quite clear from the name `output_cout_` that it has to be a member variable, I wonder if it then is too verbose to also add `this` to indicate the same? For a public member variable we would not use a trailing underscore, right? Then it would definitively be cool to add `this` to clarify (since now the trailing underscore is missing).. but is it needed for a private member variable (with trailing underscore) ?

Reply:
Well; for good and for bad we do not have any style guides which people really follow in this codebase. Personally I *always* use `this->` in code I write - and like that. For this particular case the variable `output_cout` is used as function/constructor argument in many places - I therefor think concious use of `this->`will increase readability much more than just the trailing underscore `_`.

Reply:
> For this particular case the variable output_cout is used as function/constructor argument in many places

I could only see that it is used in the constructor (as an argument). The other places in the file it is used as a member variable. And the usage as an argument in the constructor should not count since the constructor is special and it should not be confusing to use both `output_cout` as an input argument and and then initialize `output_cout_` from `output_cout` since this is commonly what a constructor does?

Anyway, you are the boss so I will follow your suggestion and update the PR. Once I get used to it, I might see that `this->output_cout_` is actually more readable than `output_cout_`